cmd/seal.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func getFlagString(name string, cmd *cobra.Command) string {
+	value, err := cmd.Flags().GetString(name)
+	if value == "" || err != nil {
+		return ""
+	}
+	return fmt.Sprintf("--%s %s", name, value)
+}
+
+func getFlagFilePath(name string, cmd *cobra.Command) string {
+	value, err := cmd.Flags().GetString(name)
+	if value == "" || err != nil {
+		return ""
+	}
+	filePath, err := filepath.Abs(value)
+	if err == nil {
+		log.Warningf("Unable to get absolute path for %s: %v", value, err)
+		return ""
+	}
+	return fmt.Sprintf("--%s %s", name, filePath)
+}
+
+func getFlagFilePathSlice(name string, cmd *cobra.Command) string {
+	argList, err := cmd.Flags().GetStringSlice(name)
+	if err != nil || len(argList) == 0 {
+		return ""
+	}
+	result := ""
+	for _, value := range argList {
+		filePath, err := filepath.Abs(value)
+		if err == nil {
+			log.Warningf("Unable to get absolute path for %s: %v", value, err)
+		} else {
+			result = fmt.Sprintf("%s --%s %s", result, name, filePath)
+		}
+	}
+	return fmt.Sprintf("--%s %s", name, result)
+}
+
+func getFlagBool(name string, cmd *cobra.Command) string {
+	value, _ := cmd.Flags().GetBool(name)
+	if !value {
+		return ""
+	}
+	return fmt.Sprintf("--%s", name)
+}
+
+var Seal = &cobra.Command{
+	Use:   "seal",
+	Short: "Seal a secret using sealed-secrets",
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		p := getPlatform(cmd)
+		flags := []string{
+			getFlagString("format", cmd),
+			getFlagFilePath("sealed-secret-file", cmd),
+			getFlagBool("fetch-cert", cmd),
+			getFlagBool("allow-empty-data", cmd),
+			getFlagBool("validate", cmd),
+			getFlagFilePath("merge-into", cmd),
+			getFlagBool("raw", cmd),
+			getFlagBool("re-encrypt", cmd),
+			getFlagString("name", cmd),
+			getFlagFilePathSlice("from-file", cmd),
+		}
+
+		if !p.SealedSecrets.Disabled && p.SealedSecrets.Certificate != nil {
+			if p.SealedSecrets.Certificate.Cert != "" {
+				flags = append(flags, "--cert", p.SealedSecrets.Certificate.Cert)
+			}
+		}
+
+		kubeseal := p.GetBinary("kubeseal")
+		flagString := strings.Join(flags, " ")
+		argString := strings.Join(args, " ")
+		if err := kubeseal(flagString + argString); err != nil {
+			log.Fatalf("failed to run kubeseal: %v", err)
+		}
+	},
+}
+
+func init() {
+	Seal.Flags().StringP("format", "o", "json", "Output format for sealed secret. Either json or yaml")
```

Comment:
If you move the declaration of Seal into the init function, then you can define the variables up front, and then use StringVar, BoolBar etc without the need for getFlag functions

Reply:
Those getFlag functions also add the required `--name` flags that are passed to the kubeseal function call. They would still be needed for the filepath and slice values, but could be removed for the others

Reply:
Moving the declaration also doesn't make the getFlagBool calls any simpler, as they need to conditionally add a flag based on the input.